Tests/Realm.Tests/Database/AAAATests.cs

```diff
@@ -0,0 +1,245 @@
+        }
+
+        private static string GetDebugView(Expression exp)
+        {
+            if (exp == null)
+            {
+                return null;
+            }
+
+            var propertyInfo = typeof(Expression).GetProperty("DebugView", BindingFlags.Instance | BindingFlags.NonPublic);
+            return propertyInfo.GetValue(exp) as string;
+        }
+
+        [Test]
+        public void WhereNullEqTest()
+        {
+            var addressEqNull = _realm.All<Person>().Where(p => p.OptionalAddress == null).ToList();
+            Assert.That(addressEqNull.Count, Is.EqualTo(1));
+            Assert.That(addressEqNull[0].FullName, Is.EqualTo("Peter Jameson"));
+        }
+
+        [Test]
+        public void WhereBooleanEqTest()
+        {
+            var isInterestingTrue = _realm.All<Person>().Where(p => p.IsInteresting == true).ToList();
+            Assert.That(isInterestingTrue.Count, Is.EqualTo(2));
+            Assert.That(isInterestingTrue[0].FullName, Is.EqualTo("John Smith"));
+            Assert.That(isInterestingTrue[1].FullName, Is.EqualTo("Peter Jameson"));
+        }
+
+        [Test]
+        public void WhereBooleanNeqTest()
+        {
+            var isInterestingFalse = _realm.All<Person>().Where(p => !p.IsInteresting).ToList();
+            Assert.That(isInterestingFalse.Count, Is.EqualTo(1));
+            Assert.That(isInterestingFalse[0].FullName, Is.EqualTo("John Doe"));
+        }
+
+        [Test]
+        public void WhereDoubleEqualTest()
+        {
+            var latitudeEq = _realm.All<Person>().Where(p => p.Latitude == 51.508530).ToList();
+            Assert.That(latitudeEq.Count, Is.EqualTo(1));
+            Assert.That(latitudeEq[0].FullName, Is.EqualTo("John Smith"));
+        }
+
+        [Test]
+        public void WhereFloatEqualTest()
+        {
+            var scoreEq = _realm.All<Person>().Where(p => p.Score == 100.0f).ToList();
+            Assert.That(scoreEq.Count, Is.EqualTo(1));
+            Assert.That(scoreEq[0].FullName, Is.EqualTo("John Doe"));
+        }
+
+        [Test]
+        public void WhereFloatNotEqualTest()
+        {
+            var scoreNeq = _realm.All<Person>().Where(p => p.Score != 100).ToList();
+            Assert.That(scoreNeq.Count, Is.EqualTo(2));
+            Assert.That(scoreNeq[0].FullName, Is.EqualTo("John Smith"));
+            Assert.That(scoreNeq[1].FullName, Is.EqualTo("Peter Jameson"));
+        }
+
+        [Test]
+        public void WhereFloatGtAndLtThanTest()
+        {
+            var scoreGtAndLt = _realm.All<Person>().Where(p => p.Score > 40
+            && p.Score < 48).ToList();
+            Assert.That(scoreGtAndLt.Count, Is.EqualTo(1));
+            Assert.That(scoreGtAndLt[0].FullName, Is.EqualTo("Peter Jameson"));
+        }
+
+        [Test]
+        public void WhereFloatGteAndLteThanTest()
+        {
+            var scoreGteAndLte = _realm.All<Person>().Where(p => p.Score >= 100.0f
+            && p.Score <= 100.0f).ToList();
```

Comment:
We try not to break the body of methods if the line is too long. You can put the whole Where on the next line if you eed to